server/controllers/chatController.js

```diff
@@ -0,0 +1,221 @@
+const mongoose = require("mongoose");
+const createError = require("http-errors");
+const moment = require("moment");
+const Chat = require("../models/chatModel");
+const Message = require("../models/messageModel");
+const Profile = require("../models/profileModel");
+const User = require("../models/userModel");
+
+exports.createChat = async (req, res, next) => {
+  try {
+    await Chat.findOne(
+      {
+        participants: ["601851731532dd4f585fced2", req.user.id], //1st PARTICIPANT - should come form body.req!!
+      },
+      async (err, chat) => {
+        if (err) console.log(err);
+        if (chat) {
+          return;
+        }
+        if (!chat) {
```

Comment:
So the intention here is that if I hire a sitter twice, there's still only one conversation between us?

Reply:
Yes. We check if they already have conversation and if they do - we don't need to create one. This would trigger when user presses "Message" button (probably on request, after it was accepted - still to be implemented though), and with existing chat they should be redirected straight to messaging with the second user